install/operator/pkg/syndesis/versions/versions.go

```diff
@@ -0,0 +1,331 @@
+		}
+
+		// Server
+		// TODO: Migrate resources
+		if api.v1alpha1.Spec.Components.Server.Features.ManagementUrlFor3scale != "" {
+			api.v1beta1.Spec.Components.Server.Features.ManagementUrlFor3scale = api.v1alpha1.Spec.Components.Server.Features.ManagementUrlFor3scale
+		}
+
+		// Database
+		// TODO: Migrate resources
+		if api.v1alpha1.Spec.Components.Db.Database != "" {
+			api.v1beta1.Spec.Components.Database.Name = api.v1alpha1.Spec.Components.Db.Database
+		}
+
+		if api.v1alpha1.Spec.Components.Db.User != "" {
+			api.v1beta1.Spec.Components.Database.User = api.v1alpha1.Spec.Components.Db.User
+		}
+
+		// Oauth
+		if api.v1alpha1.Spec.Components.Oauth.DisableSarCheck != nil {
+			api.v1beta1.Spec.Components.Oauth.DisableSarCheck = *api.v1alpha1.Spec.Components.Oauth.DisableSarCheck
+		}
+
+		// Meta
+		// TODO: Migrate resources
+
+		// UI
+		// TODO: Migrate resources
+
+		// S2I
+		// TODO: Migrate resources
+
+		// Prometheus
+		// TODO: Migrate resources
+
+		// Grafana
+		// TODO: Migrate resources
+
+		// Komodo
+		// TODO: Migrate resources
+
+		// Upgrade
+		// TODO: Migrate resources
+
+		// General
+		if api.v1alpha1.Spec.RouteHostname != "" {
+			api.v1beta1.Spec.RouteHostname = api.v1alpha1.Spec.RouteHostname
+		}
+
+		if api.v1alpha1.Spec.ImageStreamNamespace != "" {
+			api.v1beta1.Spec.ImageStreamNamespace = api.v1alpha1.Spec.ImageStreamNamespace
+		}
+
+		if api.v1alpha1.Spec.SarNamespace != "" {
+			api.v1beta1.Spec.Components.Oauth.SarNamespace = api.v1alpha1.Spec.SarNamespace
+		}
+
+		// We dont want to migrate again more than once
+		api.v1beta1.Spec.ForceMigration = false
+		api.v1beta1.Status.ApiVersion = v1beta1.SchemeGroupVersion.Version
+
+		// We need the same status and version in the target as in the origin
+		api.v1beta1.Status.Version = api.v1alpha1.Status.Version
+		api.v1beta1.Status.Phase = v1beta1.SyndesisPhaseInstalled
+		api.v1beta1.Status.Reason = v1beta1.SyndesisStatusReasonMigrated
+		api.v1beta1.Status.Description = fmt.Sprintf("App migrated from %s to %s", v1alpha1.SchemeGroupVersion.String(), v1beta1.SchemeGroupVersion.String())
+	}
+
+	return nil
+}
+
+// Write back apis
+func (api syndesisApi) updateApis() error {
+	api.log.Info("updating syndesis api",
+		"from name", api.v1alpha1.Name, "from version", api.v1alpha1.Status.Version,
+		"to name", api.v1beta1.Name, "to version", api.v1beta1.Status.Version)
+	if _, _, err := util.CreateOrUpdate(api.context, api.client, api.v1beta1); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Attempt to convert from unstructured to v1beta1.Syndesis
+func (api syndesisApi) unstructuredToV1Beta1(obj unstructured.Unstructured) (s *v1beta1.Syndesis, err error) {
+	s = &v1beta1.Syndesis{}
+
+	objM, err := json.Marshal(obj.Object)
+	if err != nil {
+		return
+	}
+
+	if err := json.Unmarshal(objM, s); err != nil {
+		return s, err
+	}
+
+	if s.Status.ApiVersion == "v1beta1" {
```

Comment:
I think there can be problems with this strategy. If someone creates a syndesis resource without that flag in the status it's interpreted as `v1alpha1` while it isn't. Especially, if you enable the status subresource, spec and status will need to be changed in two different calls and that can cause problems.

I would stay with it if it's the only option, but would try to distinguish them based on the `APIVersion` field if possible.

Reply:
Not possible, see my previous comment. Basically `APIVersion` is set to `v1beta1` even if it is a `v1alpha1`. Is that supposed to happen? I was very surprised to see that happening. See what I mean [here](https://pastebin.com/00NFu9MB)

Reply:
resolved with https://github.com/syndesisio/syndesis/pull/7636/commits/2a4f83ee4162391f430cd9c427284e09737041b6